src/errors/utils.ts

```diff
@@ -0,0 +1,19 @@
+import { EthereumRpcError } from 'eth-rpc-errors'
+
+export function handleError(e: EthereumRpcError<unknown>): void {
+  const msg = e.message
+  // checks if the user has metamask installed
+  if (msg.includes('missing provider') || msg.includes('No provider found')) {
+    throw new Error(
+      'Please install the MetaMask extension. If you are on mobile, open your MetaMask app and browse to this page.'
+    )
+  }
+
+  // ideally we would check for the error code here, but the same error code (-32000)
+  // is used for more than one error type.
+  if (msg.includes('err: insufficient funds for gas * price + value')) {
```

Comment:
I think you can also do `e.code === 'INSUFFICIENT_FUNDS'`

Reply:
Tried checking the `e.code` against https://eips.ethereum.org/EIPS/eip-1474#error-codes errors and couldn't find a "standard" response for this case :( . The code used for this is "Invalid input" (-32000) and I think it could cover other cases as well. Don't like this particular comparison (imagine if for some reason the error messages change in the future) but for now, covers this error case. Will create a new issue to enhance these custom mappings/comparisons